### DISCUSSION TITLE
Max Duration

### DISCUSSION THREAD

**Discussion by rcolepeterson** [speechly/speechly#318] [General]
I am using 
import { SpeechProvider } from "@speechly/react-client";

Is there a way to set max duration of entire conversation?
I am looking for a way to stop listening or turn off microphone if it goes on for more than a minute.
or do i do that on the admin / dashboard side?

I don't want my app to endlessly listen if someone leaves a tab open and forgets it.

Maybe i am thinking about it wrong?

**Comment by teelisyys** [comment]
Hi @rcolepeterson ! Thank you for using Speechly!

Unfortunately this is not directly supported supported by our React-client. You might implement similar functionality with a delayed call to `stop()`.

Please also note that the length of a single stream is limited to 5 minutes at the server side.

**Reply by rcolepeterson** [reply]
Thank you so much! So are you saying to me that even if i don't do anything the longest it could be is 5 minutes?

Cole

**Reply by thisismatu** [reply]
@rcolepeterson yes exactly! 👍

**Reply by rcolepeterson** [reply]
perfect

**Reply by rcolepeterson** [reply]
Sorry ... what happens at 5 minutes? Is there an error? or does it just switch from listening to !listening? is stop() just called?
THX U!!!!
